apps/next/src/popup/index.tsx

```diff
@@ -27,24 +28,56 @@ const App = lazy(() => {
 });
 
 const root = createRoot(document.getElementById('popup') as HTMLElement);
+type ProviderElement = ReactElement<
+  unknown,
+  string | React.JSXElementConstructor<any>
+>;
+
+export const Providers = ({
+  providers,
+  children,
+}: {
+  providers: ProviderElement[];
+  children: ProviderElement;
+}) => {
+  const renderProvider = (
+    renderedProviders: ProviderElement[],
+    renderedChildren: ProviderElement,
+  ) => {
+    const [provider, ...restProviders] = renderedProviders;
+
+    if (provider) {
+      return React.cloneElement(
+        provider,
+        undefined,
+        renderProvider(restProviders, renderedChildren),
+      );
+    }
+
+    return renderedChildren;
+  };
+  return renderProvider(providers, children);
+};
 
 browser.tabs.query({ active: true }).then(() => {
   root.render(
     <Sentry.ErrorBoundary>
       <Router>
-        <I18nextProvider i18n={i18next}>
-          <ConnectionContextProvider LoadingComponent={CircularProgress}>
-            <SettingsContextProvider>
-              <FeatureFlagsContextProvider>
-                <AnalyticsContextProvider>
-                  <Suspense fallback={<CircularProgress />}>
-                    <App />
-                  </Suspense>
-                </AnalyticsContextProvider>
-              </FeatureFlagsContextProvider>
-            </SettingsContextProvider>
-          </ConnectionContextProvider>
-        </I18nextProvider>
+        <Providers
+          providers={[
+            <I18nextProvider i18n={i18next} key={0} />,
+            <ConnectionContextProvider
+              LoadingComponent={CircularProgress}
+              key={1}
+            />,
+            <SettingsContextProvider key={2} />,
+            <FeatureFlagsContextProvider key={3} />,
+            <AnalyticsContextProvider key={4} />,
+            <Suspense fallback={<CircularProgress />} key={5} />,
+          ]}
```

Comment:
I remember @bartosz-the-coder-at-avalabs had a [nice idea](https://github.com/ava-labs/core-extension/pull/282#discussion_r2172536761) here about using [`Children.toArray()`](https://react.dev/reference/react/Children#children-toarray) so that `key`s are taken care of automatically.